src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,161 @@
+        user.setLogin("e@m.u");
+        user.setPassword(null);
+        user.setAge(22);
+        Assertions.assertThrows(InvalidUserCredentialsException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_emailLengthEquals6_userObjects_should_be_equals_positive() {
+        User user = new User();
+        user.setLogin("e@mail");
+        user.setPassword("test-password123");
+        user.setAge(20);
+        Assertions.assertNotNull(registrationService.register(user));
+        Assertions.assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    public void register_positiveUserEmail_userObjects_should_be_equals_positive() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("test-password123");
+        user.setAge(20);
+        Assertions.assertNotNull(registrationService.register(user));
+        Assertions.assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    public void register_nullPassword_should_throw_invalidUserCredentialsException_negative() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword(null);
+        user.setAge(22);
+        Assertions.assertThrows(InvalidUserCredentialsException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_emptyPassword_should_throw_invalidUserCredentialsException_negative() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("");
+        user.setAge(22);
+        Assertions.assertThrows(InvalidUserCredentialsException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_passwordLengthLessThan6_should_throw_exception_negative() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("123");
+        user.setAge(22);
+        Assertions.assertThrows(InvalidUserCredentialsException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_passwordLengthEquals6_userObjects_should_be_equals_positive() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("123123");
+        user.setAge(20);
+        Assertions.assertNotNull(registrationService.register(user));
+        Assertions.assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    public void register_validPassword_userObjects_should_be_equals_positive() {
+        User user = new User();
+        user.setLogin("test-mail1@gmail.com");
+        user.setPassword("password123");
+        user.setAge(20);
+        Assertions.assertNotNull(registrationService.register(user));
+        Assertions.assertEquals(user, registrationService.register(user));
+    }
+
+}
```

Comment:
Consider adding a test case to verify that registering a user with a duplicate login throws an exception or fails as expected. This aligns with the task requirement to ensure that 'there is no user with such login in the Storage yet'.